.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -240,4 +242,63 @@ jobs:
       
       - name: Inspect image
         run: |
-          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
\ No newline at end of file
+          docker buildx imagetools inspect ${{ vars.AWS_ECR_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ steps.meta.outputs.version }}
+
+  vulnerability_scan:
+    runs-on: ubuntu-latest
+    # if: inputs.publish
+    needs:
+    - docker_build
+    - prepare-metadata
+    steps:
+    - name: Create Trivy config
+      run: |
+        cat <<EOF > /tmp/trivy.yaml
+        format: sarif
+        exit-code: '1'
+        output: trivy-results.sarif
+        vulnerability:
+          ignore-unfixed: true  
+          type: 
+            - os
+            - library
+        severity: 
+          - CRITICAL
+          - HIGH
+        db:
+          repository: index.docker.io/aquasec/trivy-db:2
+        EOF
+    - name: Run Trivy vulnerability scanner
+      uses: aquasecurity/trivy-action@0.28.0
+      with:
+        image-ref: "${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}@${{ needs.docker_build.outputs.dockerhub-sha }}"
```

Comment:
Should we use `${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}:${{ needs.docker_build.outputs.dockerhub-sha }}` or `${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}@${{ needs.docker_build.outputs.dockerhub-sha }}`

Looks like the [example](https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#scan-ci-pipeline) use `:` though. Not sure about this